src/main.rs

```diff
@@ -337,10 +348,13 @@ fn handle_command_args() -> Result<(), i32> {
             } else {
                 let port = matches.value_of("port").unwrap_or("0").parse::<u16>().unwrap();
                 let provider_state = matches.value_of("provider-state")
-                  .map(|filter| Regex::new(filter).unwrap());
+                    .map(|filter| Regex::new(filter).unwrap());
+                let provider_state_header_name: String = matches.value_of
+                ("provider-state-header-name")
+                    .unwrap().into();
```

Comment:
You are unwrapping the optional here. That will result in a panic if the parameter is not supplied. Would be better to pass the optional value instead of unwrapping it.